From tklauser@distanz.ch  Tue May 11 16:36:44 2010
From: Tobias Klauser <tklauser@distanz.ch>
Date: Tue, 11 May 2010 09:10:23 +0200
Subject: serial: bfin_sport_uart: Use resource size to fix off-by-one error
To: vapier@gentoo.org, gregkh@suse.de, linux-serial@vger.kernel.org
Cc: kernel-janitors@vger.kernel.org, Tobias Klauser <tklauser@distanz.ch>
Message-ID: <1273561823-4495-1-git-send-email-tklauser@distanz.ch>


Use the resource_size function instead of manually calculating the
resource size. This actually fixes an off-by-one error.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/bfin_sport_uart.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/serial/bfin_sport_uart.c
+++ b/drivers/serial/bfin_sport_uart.c
@@ -774,8 +774,7 @@ static int __devinit sport_uart_probe(st
 			goto out_error_free_peripherals;
 		}
 
-		sport->port.membase = ioremap(res->start,
-			res->end - res->start);
+		sport->port.membase = ioremap(res->start, resource_size(res));
 		if (!sport->port.membase) {
 			dev_err(&pdev->dev, "Cannot map sport IO\n");
 			ret = -ENXIO;
